src/main/kotlin/com/feedback/resource/FeedbackResource.kt

```diff
@@ -0,0 +1,52 @@
+package com.feedback.resource
+
+import com.feedback.dao.FeedbackCreationRequest
+import com.feedback.dao.LoggedInUser
+import com.feedback.service.EmployeeFetcher
+import com.feedback.service.FeedbackInserter
+import jakarta.ws.rs.Consumes
+import jakarta.ws.rs.POST
+import jakarta.ws.rs.Path
+import jakarta.ws.rs.Produces
+import jakarta.ws.rs.core.Context
+import jakarta.ws.rs.core.Cookie
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.core.Response.Status
+import org.springframework.stereotype.Component
+
+@Component
+@Path("/api/feedback")
+@Produces("application/json")
+@Consumes("application/json")
+class FeedbackResource(
+    private val feedbackInserter: FeedbackInserter,
+    private val employeeFetcher: EmployeeFetcher,
+    private val loginDetailValidator: LoginDetailValidator,
+    private val cookiesDataExtractor: CookiesDataExtractor
+) {
+
+    @POST
+    fun submitFeedback(
+        feedbackRequest: FeedbackRequest,
```

Comment:
what's the difference between feedbackRequest and FeedbackCreationRequest? the FeedbackCreationRequest should be the input in the json. If you need to pass the department as a parameter, you can add it actually to the loggedInUser, and extend it to be not only id and companyId, but with some details about the emploee